test/spec/bidmanager_spec.js

```diff
@@ -6,300 +6,345 @@ var assert = require("assert");
 var utils = require('../../src/utils');
 var bidmanager = require('../../src/bidmanager');
 
-    describe('replaceTokenInString', function(){
+describe('replaceTokenInString', function () {
+
+  it('should replace all given tokens in a String', function () {
+    var tokensToReplace = {
+      'foo': 'bar',
+      'zap': 'quux'
+    };
+
+    var output = utils.replaceTokenInString("hello %FOO%, I am %ZAP%", tokensToReplace, "%");
+    assert.equal(output, "hello bar, I am quux");
+  });
+
+  it('should ignore tokens it does not see', function () {
+    var output = utils.replaceTokenInString("hello %FOO%", {}, "%");
+
+    assert.equal(output, "hello %FOO%");
+  });
+});
+
+describe('bidmanager.js', function () {
+
+  describe('getKeyValueTargetingPairs', function () {
+    var bid = {};
+    var bidPriceCpm = 5.578;
+    var bidPbLg = 5.50;
+    var bidPbMg = 5.50;
+    var bidPbHg = 5.57;
+    var bidPbAg = 5.50;
```

Comment:
The file changes are here and below to add the "auto" default bucket, so only a few lines, and then applied code style which updated most everything else.